dodiscover/constraint/skeleton.py

```diff
@@ -0,0 +1,470 @@
+import logging
+from collections import defaultdict
+from itertools import combinations
+from typing import Any, Dict, Iterable, List, Set
+
+import networkx as nx
+import numpy as np
+
+from dodiscover.ci import BaseConditionalIndependenceTest
+
+from ..context import Context
+
+logger = logging.getLogger()
+
+ACCEPTED_MARKOVIAN_SKELETON_METHODS = ["complete", "neighbors", "neighbors_path"]
+
+
+def _iter_conditioning_set(
+    possible_variables: Iterable,
+    x_var: Any,
+    y_var: Any,
+    size_cond_set: int,
+):
```

Comment:
I noted that there no return types (here or at other places). Maybe you can add them as well, e.g., ```-> List[Any]```.

Reply:
Added `Iterable[Set]`